icu4c/source/i18n/calendar.cpp

```diff
@@ -870,7 +870,28 @@ Calendar::createInstance(const TimeZone& zone, UErrorCode& success)
 Calendar* U_EXPORT2
 Calendar::createInstance(const Locale& aLocale, UErrorCode& success)
 {
-    return createInstance(TimeZone::createDefault(), aLocale, success);
+    Calendar* res = createInstance(TimeZone::createDefault(), aLocale, success);
+    res->adoptTimeZoneFromLocale(aLocale, success);
+    return res;
+}
+
+// ------------------------------------- Adopting
+
+void
+Calendar::adoptTimeZoneFromLocale(const Locale& aLocale, UErrorCode &status) {
+    if (U_FAILURE(status)) {
+        return;
+    }
+    char buffer[ULOC_KEYWORDS_CAPACITY] = "";
+    int32_t count = aLocale.getKeywordValue("timezone", buffer, sizeof(buffer), status);
+    if (U_FAILURE(status) || status == U_STRING_NOT_TERMINATED_WARNING) {
```

Comment:
it is better practice to make `localStatus`, rather than setting `status` and unsetting it later, for two reasons:

1. Resetting `status` to `U_ZERO_ERROR` might delete warning statuses
2. People have debuggers that look for changes to their `status` variable, and touching it will trigger their debugger

Reply:
DONE